hack/run-kiali.sh

```diff
@@ -824,25 +825,11 @@ cleanup_and_exit() {
   kill_port_forward_prometheus
   kill_port_forward_grafana
   kill_port_forward_tracing
-  restore_original_context
 
   exitmsg "Exiting"
   exit ${1:-0}
 }
 
-restore_original_context() {
```

Comment:
I'm looking through this PR now and trying to figure out what its doing and why :)

But I wonder why this is being removed - if you do not use the current context, that means this script is going to create one and then switch to it. But when the script exits, you want to put the user back to what his context originally was... otherwise, the user will be confused why he isn't back to the way things worked before he started the script.

Reply:
I think i see it - you are using a completely different kubeconfig file.. using the user's own and its current context won't change.

Reply:
> I think i see it - you are using a completely different kubeconfig file.. using the user's own and its current context won't change.

Yes rather than changing the user's kubeconfig, this change either uses their existing kubeconfig as it is or creates a new one using the kiali SA creds but either way the existing kubeconfig isn't changed. Since the "remote secret" takes a kubeconfig file as input, this seemed easier and doesn't require messing with the user's kubeconfig at all.